code/modules/mafia/controller.dm

```diff
@@ -592,10 +597,35 @@
 						continue
```

Comment:
Semi unrelated to the current code, but you should add an admin check for the admin controls. never trust the user and all

Reply:
doesn't checking for holder work fine?

Reply:
usr should be the client that caused the proc call, it can be a bit odd but it shouldn't be spoofable

Reply:
then no issue with people being able to get into admin controls.